.github/workflows/release.yaml

```diff
@@ -0,0 +1,140 @@
+name: Release primaza
+
+on:
+  push:
+    tags:
+    - 'v*' # Push events to matching v*, i.e. v1.0, v20.15.10
+  workflow_dispatch:
+    inputs:
+      version:
+        description: "Primaza release version"
+        required: true
+        default: "latest"
+
+permissions:
+  contents: write
+
+env:
+  GO111MODULE: "on"
+  GO_VERSION: "^1.20"
+  OUT_DIR: ./out/workflow
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
+
+jobs:
+  build:
+    name: Release Primaza
+    runs-on: ubuntu-20.04
+    permissions:
+      contents: write
+      packages: write
+
+    steps:
+      - name: Set up Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: ${{ env.GO_VERSION }}
+
+      - name: Set Version and Manifests
+        run: |
+          # set version based on dispatch type
+          version="latest"
+          if [[ "${{github.event_name}}" == "workflow_dispatch" ]]; then
+            version=${{ github.event.inputs.version }}
+          elif [[ "${{github.ref_type}}" == "tag" ]]; then
+            version=${{ github.ref_name }}
+          fi
+          echo "VERSION=$version" >> $GITHUB_ENV
+          echo "TENANT_CONFIG_FILE=tenant_config_$version.yaml" >> $GITHUB_ENV
+          echo "CRDS_CONFIG_FILE=crds_config_$version.yaml" >> $GITHUB_ENV
+          echo "APPLICATION_NAMESPACE_CONFIG_FILE=application_agent_config_$version.yaml" >> $GITHUB_ENV
+          echo "SERVICE_NAMESPACE_CONFIG_FILE=service_agent_config_$version.yaml" >> $GITHUB_ENV
+
+      - name: 'Log in to the Container registry'
+        uses: docker/login-action@465a07811f14bebb1938fbed4728c6a1ff8901fc
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Setup Output folder
+        run: mkdir -p ${{ env.OUT_DIR }}
+
+      - name: Delete previous release and tag
+        if: ${{ env.VERSION == 'latest' }}
+        uses: dev-drprasad/delete-tag-and-release@v0.2.1
+        with:
+          delete_release: true # default: false
+          tag_name: ${{ env.VERSION }} # tag name to delete
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
```

Comment:
I don't know if deleting the tag is a good idea.  Once created, tags are going to get pulled by anyone with a repository clone, and once tags have been pulled by those clones, they can't really be deleted.

For more information, see [git's documentation on re-tagging](https://git-scm.com/docs/git-tag#_on_re_tagging).